src/heliaServer.ts

```diff
@@ -133,44 +147,69 @@ export class HeliaServer {
   }
 
   async fetchWithoutSubdomain ({ request, reply }: RouteHandler): Promise<void> {
+    const opController = new AbortController()
+    request.raw.on('close', () => {
+      if (request.raw.aborted) {
+        this.log('Request aborted by client')
+        opController.abort()
+      }
+    })
     const { ns: namespace, address, '*': relativePath } = request.params as EntryParams
     try {
       await this.isReady
-      await this._fetch({ request, reply, address, namespace, relativePath })
+      await this._fetch({ request, reply, address, namespace, relativePath, signal: opController.signal })
     } catch (error) {
       this.log('Error requesting content from helia:', error)
       await reply.code(500).send(error)
     }
   }
 
-  async _fetch ({ reply, address, namespace, relativePath }: RouteHandler & ParsedEntryParams): Promise<void> {
+  async _fetch ({ reply, address, namespace, relativePath, signal }: RouteHandler & ParsedEntryParams & { signal: AbortSignal }): Promise<void> {
     this.log('Fetching from Helia:', { address, namespace, relativePath })
     let type: string | undefined
     // raw response is needed to respond with the correct content type.
-    for await (const chunk of await this.heliaFetch.fetch({ address, namespace, relativePath })) {
-      if (type === undefined) {
-        type = await parseContentType({ bytes: chunk, path: relativePath })
-        // this needs to happen first.
-        reply.raw.writeHead(200, {
-          'Content-Type': type ?? DEFAULT_MIME_TYPE,
-          'Cache-Control': 'public, max-age=31536000, immutable'
-        })
+    try {
+      for await (const chunk of await this.heliaFetch.fetch({ address, namespace, relativePath, signal })) {
+        if (type === undefined) {
+          type = await parseContentType({ bytes: chunk, path: relativePath })
+          // this needs to happen first.
+          reply.raw.writeHead(200, {
+            'Content-Type': type ?? DEFAULT_MIME_TYPE,
+            'Cache-Control': 'public, max-age=31536000, immutable'
+          })
+        }
+        reply.raw.write(Buffer.from(chunk))
```

Comment:
wrapped all this in a try-catch to handle bubbling errors on request aborts (and other things that fail, like some webrtc failures I posted as comments in https://github.com/ipfs/helia-http-gateway/issues/18)